### PR TITLE
normalize attribute getter behaviour

### PR DESCRIPTION
**Pull request by bcoca**
ensure they all setattr and return the value arrived at
  also added comments for future bcoca

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

playbook

### CODE REVIEW

File: lib/ansible/playbook/attribute.py [hunk 1]
```diff
@@ -142,61 +138,67 @@ def __set__(self, obj, value):
...
-                    # _get_attr_connection has been replaced by ConnectionFieldAttribute.
-                    # Leaving it here for test_attr_method from
-                    # test/units/playbook/test_base.py to pass and for backwards compat.
-                    if getattr(obj, '_squashed', False):
-                        value = getattr(obj, f'_{self.name}', Sentinel)
-                    else:
-                        value = getattr(obj, method)()
-                else:
-                    value = getattr(obj, f'_{self.name}', Sentinel)
+                # just get value fomr attribute itself as normal
```
Comment (mkrizek) [inline lib/ansible/playbook/attribute.py]:
```suggestion
                # just get value from attribute itself as normal
```

File: lib/ansible/playbook/attribute.py [hunk 2]
```diff
@@ -142,61 +138,67 @@ def __set__(self, obj, value):
...
+                value = super(FieldAttribute, self).__get__(obj, obj_type=obj_type)
```
Comment (mkrizek) [inline lib/ansible/playbook/attribute.py]:
```suggestion
                value = super().__get__(obj, obj_type=obj_type)
```

File: lib/ansible/playbook/attribute.py [hunk 3]
```diff
@@ -142,61 +138,67 @@ def __set__(self, obj, value):
     def __delete__(self, obj):
         delattr(obj, f'_{self.name}')
 
+    def _set_default(self, obj):
+        ''' handles value/callable defauts '''
```
Comment (mkrizek) [inline lib/ansible/playbook/attribute.py]:
```suggestion
        ''' handles value/callable defaults '''
```

**Comment by mkrizek** [comment]
Doing `setattr`s in `__get__`s does not feel right. 

Just to note why the only situation we do that now in `devel` is for when `default` is a `callable` and the reason for that is to allow for expressions like:

```python
task.args['_raw_params'] = 'noop'
```

(for the case when `args` is not yet defined, so it is effectively `Sentinel`)

In the above example `__set__` is not called, only `__get__` is called on `args` and `noop` is stored into whatever `__get__` returned.

So if we just return `default()`, a new `dict` would be created and `noop` would be stored into the `_raw_params` key. But the problem is that the new dict would be lost since it isn't stored on the `Attribute`.

So we also need to do `value = default(); setattr(obj, '_args', value); return value` in the `__get__`.

This program demonstrates the above:
```python
import dis

from ansible.playbook.attribute import FieldAttribute


class T:
    d = FieldAttribute(isa='dict', default=dict)
    i = FieldAttribute(isa='int', default=3)


def f():
    t = T()
    t.d['_raw_params'] = 'noop'
    t.i = 4


dis.dis(f)
print()
f()
```

a patch for `attribute.py` to see which methods are called:
```diff
diff --git a/lib/ansible/playbook/attribute.py b/lib/ansible/playbook/attribute.py
index b28405d2430..2222eb67ab9 100644
--- a/lib/ansible/playbook/attribute.py
+++ b/lib/ansible/playbook/attribute.py
@@ -132,6 +132,7 @@ class Attribute:
         return value

     def __set__(self, obj, value):
+        print(f'Attribute.__set__ {self.name} called')
         setattr(obj, f'_{self.name}', value)
         if self.alias is not None:
             setattr(obj, f'_{self.alias}', value)
@@ -155,6 +156,7 @@ class FieldAttribute(Attribute):
         self.prepend = prepend

     def __get__(self, obj, obj_type=None):
+        print(f'FieldAttribute.__get__ {self.name} called')
         if getattr(obj, '_squashed', False) or getattr(obj, '_finalized', False):
             value = getattr(obj, f'_{self.name}', Sentinel)
         else:
```

and the output:
```
 12           0 LOAD_GLOBAL              0 (T)
              2 CALL_FUNCTION            0
              4 STORE_FAST               0 (t)

 13           6 LOAD_CONST               1 ('noop')
              8 LOAD_FAST                0 (t)
             10 LOAD_ATTR                1 (d)
             12 LOAD_CONST               2 ('_raw_params')
             14 STORE_SUBSCR

 14          16 LOAD_CONST               3 (4)
             18 LOAD_FAST                0 (t)
             20 STORE_ATTR               2 (i)
             22 LOAD_CONST               0 (None)
             24 RETURN_VALUE

FieldAttribute.__get__ d called
Attribute.__set__ i called
```

**Comment by bcoca** [comment]
@mkrizek i was thinking, would it make more sense if __get__ looked at _{name} first as a 'cached result'?

**Pull request closed without merging**